src/state/apis/abi/abiApi.ts

```diff
@@ -0,0 +1,39 @@
+import { createApi } from '@reduxjs/toolkit/dist/query/react'
+import { getConfig } from 'config'
+import isEmpty from 'lodash/isEmpty'
+import { logger } from 'lib/logger'
+
+import { BASE_RTK_CREATE_API_CONFIG } from '../const'
+
+const moduleLogger = logger.child({ namespace: ['abiApi'] })
+
+type ContractAddress = string // 0xaddress on evm mainnet
+type Abi = any // json
+
+export const abiApi = createApi({
```

Comment:
this currently is ethereum mainnet specific only - we could generalize this to accept a shapeshift `chainId` such that we can call out to other explorers and get abis for contracts on other chains, e.g. https://polygonscan.com/address/0x2791bca1f2de4661ed88a30c99a7a9449aa84174#code

Reply:
this should probably require/take a `chainId` and throw if it's not an EVM chain we already support